torch/csrc/jit/passes/onnx/fixup_onnx_loop.cpp

```diff
@@ -68,8 +70,172 @@ void FixupONNXLoops(Block* block) {
+
+          // Split the added scan_output back to expected tensor sequence.
+          auto loop_output = loop_node->output(i-2);
+          Node* split_node = loop_node->owningGraph()->create(onnx::SplitToSequence);
+          loop_output->replaceAllUsesWith(split_node->output());
+          split_node->i_(attr::keepdims, 0);
+          split_node->addInput(loop_output);
+          split_node->insertAfter(loop_node);
+          split_node->output()->copyMetadata(loop_output);
+
+          // Update loop output metadata.
+          loop_output->copyMetadata(inserted_value);
+          loop_output->setType(c10::unshapedType(loop_output->type()));
+
+          // The node that produces sequence should be safe to remove now.
+          seq_node->destroy();
+
+          idx_to_remove.push_back(i);
+        }
+      }
+
+      for (size_t i = 0; i < idx_to_remove.size(); ++i) {
+        size_t idx = idx_to_remove[i] - i;
+
+        sub_block->eraseInput(idx);
+        loop_node->removeInput(idx);
+
+        // Swap output order. Move all scan outputs to the back.
+        sub_block->return_node()->addInput(sub_block->return_node()->inputs().at(idx - 1));
+        sub_block->return_node()->removeInput(idx - 1);
+
+        auto loop_out = loop_node->addOutput();
+        loop_out->copyMetadata(loop_node->outputs().at(idx - 2));
+        loop_node->outputs().at(idx - 2)->replaceAllUsesWith(loop_out);
+        loop_node->eraseOutput(idx - 2);
+      }
+
+    }
+
+  }
+}
+
+static void fuseSequenceSplitConcat(Block *b) {
+  for (auto it = b->nodes().begin(), end = b->nodes().end(); it != end; ++it) {
+    for (auto* child_block : it->blocks()) {
+      fuseSequenceSplitConcat(child_block);
+    }
+    if (it->kind() == onnx::ConcatFromSequence &&
+        it->input()->node()->kind() == onnx::SplitToSequence) {
+      if (it->input()->uses().size() > 1) {
+        continue;
+      }
+
+      auto split_node = it->input()->node();
+      auto concat_node = *it;
+
+      const auto split_axis = split_node->hasAttribute(attr::axis) ? split_node->i(attr::axis) : 0;
+      const auto split_keepdims = split_node->hasAttribute(attr::keepdims) ? split_node->i(attr::keepdims) : 1;
+      const auto concat_axis = concat_node->i(attr::axis);
+      const auto concat_new_axis = concat_node->hasAttribute(attr::new_axis) ? concat_node->i(attr::new_axis) : 0;
+      const bool has_input_split = split_node->inputs().size() == 2;
+
+      if (has_input_split) {
+        continue;
+      }
+
+      if (split_keepdims == concat_new_axis) {
+        continue;
+      }
+
+      if (split_axis != concat_axis) {
+        continue;
+      }
+
+      concat_node->output()->replaceAllUsesWith(split_node->input());
+    }
+  }
+}
+
 void FixupONNXLoops(std::shared_ptr<Graph>& graph) {
   FixupONNXLoops(graph->block());
+  ConvertSequenceDependencies(graph->block());
+  fuseSequenceSplitConcat(graph->block());
```

Comment:
nit: start function name with upper case for consistency

Reply:
thanks